src/triage/experiments/multicore.py

```diff
@@ -265,26 +265,53 @@ def test_and_evaluate(
             evaluator = evaluator_factory(db_engine=db_engine)
             individual_importance = indiv_importance_factory(db_engine=db_engine)
 
+            #First generate predictions for the testing data
             predictions_proba = predictor.predict(
                 model_id,
                 test_store,
                 misc_db_parameters=dict(),
-                train_matrix_columns=train_matrix_columns
+                train_matrix_columns=train_store.columns(),
+                matrix_type = 'Test'
             )
+
+            #Next, generate predictions for the training data for the same time period
+            predictions_proba_train = predictor.predict(
+                model_id,
+                train_store,
+                misc_db_parameters=dict(),
+                train_matrix_columns=train_store.columns(),
+                matrix_type = 'Train'
+            )
+
             logging.info('Generating individual importances for model id %s', model_id)
             individual_importance.calculate_and_save_all_methods_and_dates(
                 model_id,
                 test_store
             )
             logging.info('Generating evaluations for model id %s', model_id)
+
+            # Calls evaluate once for the testing metrics
             evaluator.evaluate(
                 predictions_proba=predictions_proba,
                 labels=test_store.labels(),
                 model_id=model_id,
                 # for evaluation range, using first to last as of time:
                 evaluation_start_time=split_def['first_as_of_time'],
                 evaluation_end_time=split_def['last_as_of_time'],
-                as_of_date_frequency=split_def['test_as_of_date_frequency']
+                as_of_date_frequency=split_def['test_as_of_date_frequency'],
+                matrix_type="Test"
```

Comment:
Another thing regarding the 'MatrixType' class.

Since triage already stores a matrix type with the matrix metadata and therefore doesn't allow using a given matrix as both train and test, we can both save time and reduce the opportunity for inconsistencies by bundling the MatrixType earlier, in the MatrixStore.

How about the MatrixStore exposes a MatrixType for its instance? You don't even need to add anything to the constructor: just read the value from matrix metadata! Then that becomes the only place you need to have this if statement.

```
class MatrixStore
...
@property
def matrix_type(self):
   if self.metadata['matrix_type'] == 'train':
       return TrainMatrix
   elif ...
       return TestMatrix
   else:
      raise error
```

This way, you don't actually need to add a matrix_type argument to `predictor.predict`. This information is already being encoded in the matrix store. `evaluator.evaluate` doesn't currently take a matrix store as an argument, so you still have to add something to that call signature, but that value can just come from the matrix store's `.matrix_type` property.